docs/instructlab-cli-1.0.0.md

```diff
@@ -0,0 +1,35 @@
+# The Road to 1.0.0
+
+_Or: How I Learned to Stop Worrying and Love to GA_
+
+## Context and Goals
+
+At the time of this writing, the InstructLab CLI has gone from a scrappy research project to an upstream community serving as the basis for multiple downstreams, with the goal to continuing to evolve the community to encourage more participation from additional stakeholders. To wit, it would behoove us to determine what exactly we should be roadmapping between now and a proper 1.0.0 release, which demonstrates the following to existing and potential community members:
+
+1. An official goalpost for the community denoting the evolution of InstructLab from a pre-1.0 project lacking the stability and supportability typically seen from 1.0-and-beyond projects.
+1. A dedicated set of V1 interfaces that can be counted on for continuous usage of InstructLab 1.0 with future provisions made for backwards compatibility for subsequent Y-Streams and Z-Streams.
+1. A commitment from the Oversight Committee and Maintainer teams to continue to maintain InstructLab throughout a 1.y cycle and work towards an eventual 2.0.
+
+## MVP for an InstructLab CLI 1.0.0
+
+At a high-level, these are the items the Maintainer teams believe should serve as prereqs for releasing an InstructLab CLI 1.0.0:
+
+- Removal of the need for a `config.yaml` with a fully-realized configuration scheme, centered around the usage of system profiles
+- An official v1 REST API schema
+- Integration of the InstructLab CLI with RAG
+- An upgrade path to subsequent Y-Streams and an eventual 2.0
+- Backwards compatibility across the 1.y stream
+- An official hardware support matrix
+
```

Comment:
I would also add sufficient unit and integration testing from the subcomponents. That's something I care about building for `instructlab/training`. A user of a "1.0" project should be able to run most of the tests quickly on their machine and cover most behavior.

Reply:
I very much agree with the sentiment - can you think of any concrete milestones we may want to define here? I do want to keep this confined mainly to the CLI, so keeping it moreso in that context than the libraries

Reply:
Can we not say we have achieved a certain phase of the CI improvement plan before we can declare 1.0.0? (if one of the goals is stability....)

Reply:
Hmmm yes I like that, I can add a point around this!